### PR TITLE
Refactor executor utility functions into multiple files

### PR DESCRIPTION
**Pull request by marcocitus**
The adaptive executor is a complex machine, but `adaptive_executor.c` is made extra hard to follow by having a bunch of utility functions sprinkled in between the core logic. 

This PR moves the functions that do not depend on executor data structures into their own files (low hanging fruit). This mostly moves code around so the main thing to review is the file structure.

### CODE REVIEW

File: src/backend/distributed/executor/executor_util_tuples.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			}
+			else
+			{
+				nulls[i] = true;
+			}
+		}
+		else
+		{
+			/* Handle dropped attributes by setting to NULL */
+			dvalues[i] = (Datum) 0;
```
Comment (onderkalaci) [inline src/backend/distributed/executor/executor_util_tuples.c:111]: as much as I like this refactor, I find it a bit concerning that we do not have coverage for this line (and also for `if (AllModificationsCommutative)` case).

Ofc, not related to this PR, but wanted to note so that we can check later on.

File: src/backend/distributed/executor/executor_util_tuples.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/*-------------------------------------------------------------------------
+ *
+ * executor_util_tuples.c
```
Comment (onderkalaci) [inline src/backend/distributed/executor/executor_util_tuples.c:3]: very minor: we are already under `executor` folder, so might omit `executor_` in the file name   (as we don't have a specific header for this file but use executor_util.h)

**Review by onderkalaci — APPROVED** [review]

**Comment by marcocitus** [comment]
I'll defer merging until after 11.2 release in case we need executor-related backports.

**Comment by onderkalaci** [comment]
> I'll defer merging until after 11.2 release in case we need executor-related backports.

@marcocitus Maybe good time to merge nowadays?
